presto-main/src/main/java/io/prestosql/operator/BasicWorkProcessorOperatorAdapter.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.operator;
+
+import io.prestosql.execution.Lifespan;
+import io.prestosql.operator.WorkProcessorOperatorAdapter.AdapterWorkProcessorOperator;
+import io.prestosql.operator.WorkProcessorOperatorAdapter.AdapterWorkProcessorOperatorFactory;
+import io.prestosql.spi.Page;
+import io.prestosql.sql.planner.plan.PlanNodeId;
+
+import java.util.Optional;
+
+import static java.util.Objects.requireNonNull;
+
+public class BasicWorkProcessorOperatorAdapter
```

Comment:
It isn't clear when I would use this instead of `AdapterWorkProcessorOperator`.  The other direct implementations of `AdapterWorkProcessorOperator` use a buffer, but seem to have some slightly different code for things like needs input... maybe add a comment about the requirement (assumptions?) for using this class.

Reply:
added comments